src/components/Request/Pay/Views/Initial.view.tsx

```diff
@@ -144,6 +144,10 @@ export const InitialView = ({
         }
     }, [requestLinkData, tokenPriceData])
 
+    useEffect(() => {
+        resetTokenAndChain()
+    }, [])
+
```

Comment:
Style: Code wise, it's gonna be really difficult to parse in 6 months WHY we have this code here. But there's a good reason for it!

For code like this, I think adding a small comment above explaining its purpose is the right thing to do